src/java.base/share/classes/java/net/URI.java

```diff
@@ -3154,6 +3154,13 @@ private int scan(int start, int n, long lowMask, long highMask)
             return p;
         }
 
+        // parse as a Registry-based authority if not insisting
+        // upon a server-based authority
+        private boolean parseRegistryChars()
+        {
+            return requireServerAuthority?false:true;
+        }
```

Comment:
Ugh, this is very messy. Can you reformat this to be consistent with the existing code?